android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMKeyboardPickerAdapter.java

```diff
@@ -48,7 +50,25 @@ public KMKeyboardPickerAdapter(final Context context, Dataset.Keyboards adapter)
         List<Keyboard> kbdList = new ArrayList<>(kbdMapList.size());
 
         for(HashMap<String, String> kbdMap: kbdMapList) {
-          kbdList.add(new Keyboard(kbdMap));
+          boolean isCustom = kbdMap.containsKey(KMManager.KMKey_CustomKeyboard) &&
+            kbdMap.get(KMManager.KMKey_CustomKeyboard).equals("Y");
+          boolean isNewKeyboard = kbdMap.containsKey(KeyboardPickerActivity.KMKEY_INTERNAL_NEW_KEYBOARD) &&
+            kbdMap.get(KeyboardPickerActivity.KMKEY_INTERNAL_NEW_KEYBOARD).equals(KeyboardPickerActivity.KMKEY_INTERNAL_NEW_KEYBOARD);
+
+          Keyboard k = new Keyboard(
```

Comment:
This bothers me a little because we are shifting from one set of issues to another -- this introduces a very heavy constructor with a big reliance on parameter order. Very easy to get wrong. Sadly, Java doesn't have great idioms to improve this -- you could introduce a [KeyboardBuilder](https://stackoverflow.com/a/1988035/1836776), but that's not ideal either. Or set the properties one at a time... Oh for Named Parameters.

I guess what I am looking for is:
1. non-confusable property naming -- `k.setX()` achieves this
2. a way to avoid forgetting required parameters -- could be done with an assertion or similar?

Have a think about it and we can discuss at standup if you like?

Reply:
A recap of our discussion at standup:

We'll keep the long list of param names in the constructor. If we really want named parameter lists at some point, migrate to Kotlin.

Not using Builder pattern because we wouldn't catch all the changes if new parameters get added.

I'll still make the .setX() names consistent.